[sig-scheduling] deadlock/timeout running pkg/scheduler TestSchedulerMultipleProfilesScheduling

**Which test(s) are flaking**:

pkg/scheduler unit tests occasionally timeout at 5 minutes

**Reason for failure**:

Stats from 10 runs of the package:
```
Stats over 10 runs: max = 315.3s, min = 30.9s, avg = 67.0s, dev = 83.0s
```

The average (excluding the run that timed out) is ~40 seconds. The run that timed out took at least 5 minutes before being killed. That smells like a deadlock / failure-to-progress scenario which could be a real bug.

Our presubmit CI runs tolerate flakes, retrying 3 times and passing on a single success, which is masking this issue.

Observed in https://github.com/kubernetes/kubernetes/pull/93605#issuecomment-668127993 which is turning off that toleration and exercising every unit test package 10 times to ensure consistent passes:

```
//pkg/scheduler

TIMEOUT: //pkg/scheduler:go_default_test (Summary)
      /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/execroot/io_k8s_kubernetes/bazel-out/k8-fastbuild/testlogs/pkg/scheduler/go_default_test/run_8_of_10/test.log
```
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93605/pull-kubernetes-bazel-test/1290315521706692610

```
//pkg/scheduler:go_default_test                                         TIMEOUT in 1 out of 10 in 315.3s
  Stats over 10 runs: max = 315.3s, min = 30.9s, avg = 67.0s, dev = 83.0s
```
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93605/pull-kubernetes-bazel-test/1290996656451358721

```
TIMEOUT: //pkg/scheduler:go_default_test (Summary)
      /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/execroot/io_k8s_kubernetes/bazel-out/k8-fastbuild/testlogs/pkg/scheduler/go_default_test/run_5_of_10/test.log
INFO: From Testing //pkg/scheduler:go_default_test (run 5 of 10):
```

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93605/pull-kubernetes-bazel-test/1292167859186700288


I can reproduce this by building the unit test package with:
```
go test -race -c ./pkg/scheduler
```

and then running repeatedly with `stress`:

this appears to hang increasing numbers of parallel runs on the deadlock until timeouts are hit:

```
stress ./scheduler.test 
0 runs so far, 0 failures
0 runs so far, 0 failures
0 runs so far, 0 failures
0 runs so far, 0 failures
10 runs so far, 0 failures <-- 10 runs passing at a time
10 runs so far, 0 failures
10 runs so far, 0 failures
19 runs so far, 0 failures <-- 9 runs passing at a time
19 runs so far, 0 failures
19 runs so far, 0 failures
19 runs so far, 0 failures
27 runs so far, 0 failures <-- 8 runs passing at a time
27 runs so far, 0 failures
27 runs so far, 0 failures
27 runs so far, 0 failures
34 runs so far, 0 failures <-- 7 runs passing at a time
34 runs so far, 0 failures
34 runs so far, 0 failures
39 runs so far, 0 failures <-- 5 runs passing at a time
39 runs so far, 0 failures
39 runs so far, 0 failures
39 runs so far, 0 failures
44 runs so far, 0 failures <-- 5 runs passing at a time
44 runs so far, 0 failures
44 runs so far, 0 failures
44 runs so far, 0 failures
49 runs so far, 0 failures <-- 5 runs passing at a time
49 runs so far, 0 failures
49 runs so far, 0 failures
54 runs so far, 0 failures <-- 5 runs passing at a time
54 runs so far, 0 failures
54 runs so far, 0 failures
54 runs so far, 0 failures
59 runs so far, 0 failures <-- 5 runs passing at a time
59 runs so far, 0 failures
59 runs so far, 0 failures
59 runs so far, 0 failures
64 runs so far, 0 failures <-- 5 runs passing at a time
64 runs so far, 0 failures
64 runs so far, 0 failures
69 runs so far, 0 failures <-- 5 runs passing at a time
69 runs so far, 0 failures
69 runs so far, 0 failures
69 runs so far, 0 failures
74 runs so far, 0 failures <-- 5 runs passing at a time
74 runs so far, 0 failures
74 runs so far, 0 failures
74 runs so far, 0 failures
78 runs so far, 0 failures <-- 4 runs passing at a time
78 runs so far, 0 failures
78 runs so far, 0 failures
82 runs so far, 0 failures <-- 4 runs passing at a time
82 runs so far, 0 failures
82 runs so far, 0 failures
82 runs so far, 0 failures
86 runs so far, 0 failures <-- 4 runs passing at a time
86 runs so far, 0 failures
86 runs so far, 0 failures
86 runs so far, 0 failures
90 runs so far, 0 failures <-- 4 runs passing at a time
90 runs so far, 0 failures
90 runs so far, 0 failures
93 runs so far, 0 failures
94 runs so far, 0 failures <-- 4 runs passing at a time
94 runs so far, 0 failures
94 runs so far, 0 failures
97 runs so far, 0 failures <-- 3 runs passing at a time
97 runs so far, 0 failures
97 runs so far, 0 failures
97 runs so far, 0 failures
100 runs so far, 0 failures <-- 3 runs passing at a time
100 runs so far, 0 failures
100 runs so far, 0 failures
102 runs so far, 0 failures
103 runs so far, 0 failures <-- 3 runs passing at a time
103 runs so far, 0 failures
103 runs so far, 0 failures
105 runs so far, 0 failures <-- 2 runs passing at a time
105 runs so far, 0 failures
105 runs so far, 0 failures
105 runs so far, 0 failures
107 runs so far, 0 failures <-- 2 runs passing at a time
107 runs so far, 0 failures
107 runs so far, 0 failures
107 runs so far, 0 failures
108 runs so far, 0 failures <-- 1 run passing at a time
108 runs so far, 0 failures
108 runs so far, 0 failures
109 runs so far, 0 failures <-- 1 run passing at a time
109 runs so far, 0 failures
109 runs so far, 0 failures
109 runs so far, 0 failures
110 runs so far, 0 failures <-- 1 run passing at a time
110 runs so far, 0 failures
110 runs so far, 0 failures
110 runs so far, 0 failures
111 runs so far, 0 failures <-- 1 run passing at a time
111 runs so far, 0 failures
111 runs so far, 0 failures
112 runs so far, 0 failures <-- 1 run passing at a time
112 runs so far, 0 failures
112 runs so far, 0 failures
112 runs so far, 0 failures
113 runs so far, 0 failures <-- 1 run passing at a time
113 runs so far, 0 failures
113 runs so far, 0 failures
113 runs so far, 0 failures
114 runs so far, 0 failures <-- 1 run passing at a time
114 runs so far, 0 failures
114 runs so far, 0 failures
115 runs so far, 0 failures <-- 1 run passing at a time
115 runs so far, 0 failures
115 runs so far, 0 failures
115 runs so far, 0 failures
116 runs so far, 0 failures <-- 1 run passing at a time
116 runs so far, 0 failures
116 runs so far, 0 failures
116 runs so far, 0 failures
117 runs so far, 0 failures <-- 1 run passing at a time
117 runs so far, 0 failures

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200811T112642-732508107
I0811 11:26:44.292873   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.293370   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.350515   15646 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
E0811 11:26:44.350677   15646 factory.go:473] pod: default/test-pod is already present in the active queue
I0811 11:26:44.363251   15646 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
I0811 11:26:44.396250   15646 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
W0811 11:26:44.412419   15646 factory.go:467] Pod default/test-pod doesn't exist in informer cache: pod "test-pod" not found
I0811 11:26:44.456364   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.456467   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.459531   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.459617   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.463032   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.463077   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.468029   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.468077   15646 registry.go:173] Registering SelectorSpread plugin
E0811 11:26:44.477001   15646 scheduler.go:493] "Error selecting node for pod" err="scheduler" pod="foo"
E0811 11:26:44.494765   15646 framework.go:747] plugin "DefaultBinder" failed to bind pod "/foo": binder
I0811 11:26:44.497112   15646 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.497151   15646 registry.go:173] Registering SelectorSpread plugin
SIGABRT: abort
PC=0x7fff6ebc7882 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x3fb0980, 0x3fb0940, 0x7ffe00000000)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x1082f05)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/os_darwin.go:63 +0x8d
runtime

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200811T112642-114384046
I0811 11:26:44.247760   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.248161   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.295037   15645 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
E0811 11:26:44.295225   15645 factory.go:473] pod: default/test-pod is already present in the active queue
I0811 11:26:44.297640   15645 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
I0811 11:26:44.299026   15645 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
W0811 11:26:44.299086   15645 factory.go:467] Pod default/test-pod doesn't exist in informer cache: pod "test-pod" not found
I0811 11:26:44.304695   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.304749   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.307410   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.307494   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.311800   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.311852   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.324513   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.324557   15645 registry.go:173] Registering SelectorSpread plugin
E0811 11:26:44.332352   15645 scheduler.go:493] "Error selecting node for pod" err="scheduler" pod="foo"
E0811 11:26:44.351749   15645 framework.go:747] plugin "DefaultBinder" failed to bind pod "/foo": binder
I0811 11:26:44.354466   15645 registry.go:173] Registering SelectorSpread plugin
I0811 11:26:44.354543   15645 registry.go:173] Registering SelectorSpread plugin
SIGABRT: abort
PC=0x7fff6ebc7882 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x3fb0980, 0x3fb0940, 0x7ffe00000000)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x1082f05)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/os_darwin.go:63 +0x8d
runtime
119 runs so far, 2 failures
120 runs so far, 2 failures
120 runs so far, 2 failures
122 runs so far, 2 failures

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200811T112642-260875829
I0811 11:27:02.617994   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.618394   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.651476   16406 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
E0811 11:27:02.651593   16406 factory.go:473] pod: default/test-pod is already present in the active queue
I0811 11:27:02.653307   16406 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
I0811 11:27:02.654971   16406 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
W0811 11:27:02.655047   16406 factory.go:467] Pod default/test-pod doesn't exist in informer cache: pod "test-pod" not found
I0811 11:27:02.661031   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.661077   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.664505   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.664558   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.668167   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.668223   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.675560   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.675606   16406 registry.go:173] Registering SelectorSpread plugin
E0811 11:27:02.683505   16406 scheduler.go:493] "Error selecting node for pod" err="scheduler" pod="foo"
E0811 11:27:02.700017   16406 framework.go:747] plugin "DefaultBinder" failed to bind pod "/foo": binder
I0811 11:27:02.702558   16406 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:02.702607   16406 registry.go:173] Registering SelectorSpread plugin
SIGABRT: abort
PC=0x7fff6ebc7882 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x3fb0980, 0x3fb0940, 0xc000000000)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x7ffeefbff150)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/os_darwin.go:63 +0x8d
runt
123 runs so far, 3 failures
124 runs so far, 3 failures
124 runs so far, 3 failures

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200811T112642-483177488
I0811 11:27:21.197138   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.197426   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.224252   18026 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
E0811 11:27:21.224316   18026 factory.go:473] pod: default/test-pod is already present in the active queue
I0811 11:27:21.225396   18026 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
I0811 11:27:21.226301   18026 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
W0811 11:27:21.226337   18026 factory.go:467] Pod default/test-pod doesn't exist in informer cache: pod "test-pod" not found
I0811 11:27:21.231279   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.231340   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.234819   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.234877   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.238990   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.239033   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.252809   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.252854   18026 registry.go:173] Registering SelectorSpread plugin
E0811 11:27:21.267670   18026 scheduler.go:493] "Error selecting node for pod" err="scheduler" pod="foo"
E0811 11:27:21.284689   18026 framework.go:747] plugin "DefaultBinder" failed to bind pod "/foo": binder
I0811 11:27:21.286857   18026 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:21.286908   18026 registry.go:173] Registering SelectorSpread plugin
SIGABRT: abort
PC=0x7fff6ebc7882 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x3fb0980, 0x3fb0940, 0x0)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x1082f05)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/os_darwin.go:63 +0x8d
runtime.notesleep(
127 runs so far, 4 failures
127 runs so far, 4 failures
128 runs so far, 4 failures
128 runs so far, 4 failures

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200811T112642-000171567
I0811 11:27:40.042638   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.043113   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.077819   19229 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
E0811 11:27:40.077935   19229 factory.go:473] pod: default/test-pod is already present in the active queue
I0811 11:27:40.079698   19229 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
I0811 11:27:40.081473   19229 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
W0811 11:27:40.081528   19229 factory.go:467] Pod default/test-pod doesn't exist in informer cache: pod "test-pod" not found
I0811 11:27:40.087750   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.087809   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.090643   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.090688   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.094896   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.094940   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.098269   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.098336   19229 registry.go:173] Registering SelectorSpread plugin
E0811 11:27:40.102779   19229 scheduler.go:493] "Error selecting node for pod" err="scheduler" pod="foo"
E0811 11:27:40.111396   19229 framework.go:747] plugin "DefaultBinder" failed to bind pod "/foo": binder
I0811 11:27:40.113703   19229 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:40.113756   19229 registry.go:173] Registering SelectorSpread plugin
SIGABRT: abort
PC=0x7fff6ebc7882 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x3fb0980, 0x3fb0940, 0xc000000000)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x7ffeefbff150)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/os_darwin.go:63 +0x8d
runt
132 runs so far, 5 failures
132 runs so far, 5 failures
132 runs so far, 5 failures
134 runs so far, 5 failures

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200811T112642-037190594
I0811 11:27:58.094047   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.094323   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.118716   20690 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
E0811 11:27:58.118783   20690 factory.go:473] pod: default/test-pod is already present in the active queue
I0811 11:27:58.119864   20690 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
I0811 11:27:58.120776   20690 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
W0811 11:27:58.120814   20690 factory.go:467] Pod default/test-pod doesn't exist in informer cache: pod "test-pod" not found
I0811 11:27:58.124774   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.124809   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.126853   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.126884   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.129866   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.129902   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.133213   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.133243   20690 registry.go:173] Registering SelectorSpread plugin
E0811 11:27:58.137720   20690 scheduler.go:493] "Error selecting node for pod" err="scheduler" pod="foo"
E0811 11:27:58.147714   20690 framework.go:747] plugin "DefaultBinder" failed to bind pod "/foo": binder
I0811 11:27:58.149387   20690 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.149420   20690 registry.go:173] Registering SelectorSpread plugin
W0811 11:28:29.202455   20690 cache.go:722] Pod /pod1 expired
SIGABRT: abort
PC=0x7fff6ebc7882 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x3fb0980, 0x3fb0940, 0xc000000000)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x7ffeefbff150)
	/Users/lig

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200811T112642-323744825
I0811 11:27:58.478222   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.478620   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.524382   20726 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
E0811 11:27:58.524471   20726 factory.go:473] pod: default/test-pod is already present in the active queue
I0811 11:27:58.525969   20726 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
I0811 11:27:58.527589   20726 factory.go:461] "Error scheduling pod; retrying" pod="default/test-pod"
W0811 11:27:58.527644   20726 factory.go:467] Pod default/test-pod doesn't exist in informer cache: pod "test-pod" not found
I0811 11:27:58.533661   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.533710   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.536771   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.536809   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.541112   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.541167   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.545716   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.545761   20726 registry.go:173] Registering SelectorSpread plugin
E0811 11:27:58.551796   20726 scheduler.go:493] "Error selecting node for pod" err="scheduler" pod="foo"
E0811 11:27:58.562932   20726 framework.go:747] plugin "DefaultBinder" failed to bind pod "/foo": binder
I0811 11:27:58.565846   20726 registry.go:173] Registering SelectorSpread plugin
I0811 11:27:58.565896   20726 registry.go:173] Registering SelectorSpread plugin
W0811 11:28:29.627600   20726 cache.go:722] Pod /pod1 expired
SIGABRT: abort
PC=0x7fff6ebc7882 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x3fb0980, 0x3fb0940, 0xc000000000)
	/Users/liggitt/.gvm/gos/go1.15rc2/src/runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x7ffeefbff150)
	/Users/lig
138 runs so far, 7 failures
138 runs so far, 7 failures
138 runs so far, 7 failures
```

/sig scheduling

can consistently reproduce a hang with:

```
go test ./pkg/scheduler -race -timeout=10s -count=100 -v -run TestSchedulerMultipleProfilesScheduling
```

I added this debugging:
```diff
diff --git a/pkg/scheduler/scheduler_test.go b/pkg/scheduler/scheduler_test.go
index 914185bd879..72b0a7bd2d8 100644
--- a/pkg/scheduler/scheduler_test.go
+++ b/pkg/scheduler/scheduler_test.go
@@ -464,6 +464,7 @@ func TestSchedulerMultipleProfilesScheduling(t *testing.T) {
 		}
 		binding := action.(clienttesting.CreateAction).GetObject().(*v1.Binding)
 		bindings[binding.Name] = binding.Target.Name
+		fmt.Println("done create", binding.Name, binding.Target.Name)
 		wg.Done()
 		return true, binding, nil
 	})
@@ -471,9 +472,11 @@ func TestSchedulerMultipleProfilesScheduling(t *testing.T) {
 	stopFn := broadcaster.StartEventWatcher(func(obj runtime.Object) {
 		e, ok := obj.(*eventsv1.Event)
 		if !ok || e.Reason != "Scheduled" {
+			fmt.Println(obj)
 			return
 		}
 		controllers[e.Regarding.Name] = e.ReportingController
+		fmt.Println("done event", e.Regarding.Name, e.ReportingController)
 		wg.Done()
 	})
 	defer stopFn()
@@ -489,7 +492,9 @@ func TestSchedulerMultipleProfilesScheduling(t *testing.T) {
 			t.Fatal(err)
 		}
 	}
+	fmt.Println(500)
 	wg.Wait()
+	fmt.Println(502)
 
 	// Verify correct bindings and reporting controllers.
 	if diff := cmp.Diff(wantBindings, bindings); diff != "" {

```

and the hang appears to be while waiting for the create calls and resulting events to be produced:

```
=== RUN   TestSchedulerMultipleProfilesScheduling
I0811 12:21:39.208753   32752 registry.go:173] Registering SelectorSpread plugin
I0811 12:21:39.208794   32752 registry.go:173] Registering SelectorSpread plugin
500
done create pod1 machine3
done event pod1 match-machine3
done create pod2 machine2
done event pod2 match-machine2
done create pod3 machine2
done event pod3 match-machine2
panic: test timed out after 10s
FAIL	k8s.io/kubernetes/pkg/scheduler	11.323s
```

--- a/pkg/scheduler/scheduler_test.go
+++ b/pkg/scheduler/scheduler_test.go
@@ -480,6 +480,7 @@ func TestSchedulerMultipleProfilesScheduling(t *testing.T) {
 
 	// Run scheduler.
 	informerFactory.Start(ctx.Done())
+	informerFactory.WaitForCacheSync(ctx.Done())
 	go sched.Run(ctx)
 
 	// Send pods to be scheduled.

